File: app/src/main/java/wangdaye/com/geometricweather/db/controllers/DailyEntityController.java
package wangdaye.com.geometricweather.db.controllers;

import androidx.annotation.NonNull;

import java.util.List;

import wangdaye.com.geometricweather.common.basic.models.options.provider.WeatherSource;
import wangdaye.com.geometricweather.db.GeometricWeatherDatabase;
import wangdaye.com.geometricweather.db.converters.WeatherSourceConverter;
import wangdaye.com.geometricweather.db.entities.DailyEntity;

public class DailyEntityController extends AbsEntityController {

    public static void insertDailyList(@NonNull GeometricWeatherDatabase database,
                                       @NonNull List<DailyEntity> entityList) {
        if (!entityList.isEmpty()) {
            database.dailyDao().insertDailyList(entityList);
        }
    }

    public static void deleteDailyEntityList(@NonNull GeometricWeatherDatabase database,
                                             @NonNull String cityId,
                                             @NonNull WeatherSource source) {
        database.dailyDao().deleteDailyEntityList(
                cityId,
                new WeatherSourceConverter().convertToDatabaseValue(source)
        );
    }

    @NonNull
    public static List<DailyEntity> selectDailyEntityList(@NonNull GeometricWeatherDatabase database,
                                                          @NonNull String cityId,
                                                          @NonNull WeatherSource source) {
        return getNonNullList(
                database.dailyDao().selectDailyEntityList(
                        cityId,
                        new WeatherSourceConverter().convertToDatabaseValue(source)
                )
        );
    }
}

File: data/src/main/java/wangdaye/com/geometricweather/db/controllers/DailyEntityController.kt
package wangdaye.com.geometricweather.db.controllers

import wangdaye.com.geometricweather.common.basic.models.options.provider.WeatherSource
import wangdaye.com.geometricweather.db.GeometricWeatherDatabase
import wangdaye.com.geometricweather.db.converters.WeatherSourceConverter
import wangdaye.com.geometricweather.db.entities.DailyEntity

object DailyEntityController {

    @JvmStatic
    fun insertDailyList(database: GeometricWeatherDatabase, entityList: List<DailyEntity>) {
        if (entityList.isNotEmpty()) {
            database.dailyDao().insertDailyList(entityList)
        }
    }

    @JvmStatic
    fun deleteDailyEntityList(
        database: GeometricWeatherDatabase,
        cityId: String,
        source: WeatherSource
    ) {
        database.dailyDao().deleteDailyEntityList(
            cityId,
            WeatherSourceConverter().convertToDatabaseValue(source)
        )
    }

    @JvmStatic
    fun selectDailyEntityList(
        database: GeometricWeatherDatabase,
        cityId: String,
        source: WeatherSource
    ): List<DailyEntity> {
        return getNonNullList(
            database.dailyDao().selectDailyEntityList(
                cityId,
                WeatherSourceConverter().convertToDatabaseValue(source)
            )
        )
    }
}
